### PR TITLE
provider/openstack:adding metadata tags to server group create and clone

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers @derekolk @varikin

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy [hunk 1]
```diff
@@ -105,15 +115,24 @@ class ServerGroupParameters {
    */
   static List<String> unescapePythonUnicodeJsonList(String string) {
     string?.split(",")?.collect { s ->
-      s.replace("u'", "")
-        .replace("'","")
-        .replace("[","")
-        .replace("]","")
-        .replaceAll("([ ][ ]*)","")
+      s.replace("u'", "").replace("'","").replace("[","").replace("]","").replaceAll("([ ][ ]*)","")
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: Is this still needed if we are only supporting Mitaka? Or is that bug still in Mitaka?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy [hunk 2]
```diff
@@ -105,15 +115,24 @@ class ServerGroupParameters {
...
   /**
+   * Stack parameters of type 'comma_delimited_list' come back as a unicode json string. We need to split that up.
+   *
+   * To reiterate, I need a shower.
+   *
+   * @param string
+   * @return
+   */
+  static Map<String, String> unescapePythonUnicodeJsonMap(String string) {
+    String parsed = string?.replace("u'", "")?.replace("'","")?.replace("{","")?.replace("}","")?.replace("\"","")?.replaceAll("([ ][ ]*)","")
```
Comment (varikin) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: another one?!

File: clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/CloneOpenstackAtomicOperationSpec.groovy [hunk 3]
```diff
@@ -89,7 +89,8 @@ class CloneOpenstackAtomicOperationSpec extends Specification {
       securityGroups: SECURITY_GROUPS,
       autoscalingType: ServerGroupParameters.AutoscalingType.CPU,
       scaleup: scaleup,
-      scaledown: scaledown
+      scaledown: scaledown,
+      tags: ['foo':'barf']
```
Comment (derekolk) [inline clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/CloneOpenstackAtomicOperationSpec.groovy]: Let's remove this.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy [hunk 4]
```diff
@@ -105,15 +115,24 @@ class ServerGroupParameters {
    */
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: We should add a TODO and reference the openstack ticket that will address this issue in both circumstances.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy [hunk 5]
```diff
@@ -105,15 +115,24 @@ class ServerGroupParameters {
...
-        .replace("]","")
-        .replaceAll("([ ][ ]*)","")
+      s.replace("u'", "").replace("'","").replace("[","").replace("]","").replaceAll("([ ][ ]*)","")
     } ?: []
   }
 
   /**
+   * Stack parameters of type 'comma_delimited_list' come back as a unicode json string. We need to split that up.
+   *
+   * To reiterate, I need a shower.
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: Probably can remove the shower comments too.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy (hunk 1 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: This is Mitaka, and even found on the master branch of the heat api.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy (hunk 5 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/description/servergroup/ServerGroupParameters.groovy]: SO unprofessional.

**Comment by drmaas** [comment]
@spinnaker/reviewers

**Comment by duftler** [comment]
LGTM.

**Pull request merged**